app/code/Magento/CatalogUrlRewrite/Observer/CategoryUrlPathAutogeneratorObserver.php

```diff
@@ -92,6 +92,11 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         $useDefaultAttribute = !empty($category->getData('use_default')['url_key']);
         if ($category->getUrlKey() !== false && !$useDefaultAttribute) {
             $resultUrlKey = $this->categoryUrlPathGenerator->getUrlKey($category);
+            if(empty($resultUrlKey)) {
+                //NOTE: If resultUrlKey not found due to entering category name in other language then use category name
+                $resultUrlKey = $category->getName();
```

Comment:
Hi @chohanbhagchand. I would not suggest bypassing the URL filter and use the category name as it is. The `\Magento\Catalog\Model\Category::formatUrlKey` method uses the filter to make sure that the category name is properly converted to a URL key. 
We should adjust the filter logic instead if it cannot handle some Unicode names.

Reply:
@rogyar Thanks for reviewing my code. So can you please suggest the filter logic for some Unicode

Reply:
Hi @chohanbhagchand. I won't suggest an exact solution but I would recommend checking the following call deeply

```
$resultUrlKey = $this->categoryUrlPathGenerator->getUrlKey($category);
```

For some Unicode values (like strings in the Thai language) the `getUrlKey` returns an empty string. It should not happen since the URL key may not be empty. We need to have either a correct exception in this case or we need to figure out why it's empty in case of some Unicode values. At first glance, the problem is in the URL filter which cannot process Unicode values. After some debugging, you can say for sure why it happens.